draft-ietf-jsonpath-base.md

```diff
@@ -710,10 +707,11 @@ Queries:
 #### Syntax {#syntax-index}
 {: unnumbered}
 
-An index selector `[<index>]` addresses at most one object member value or at most one array element value.
+An index selector `[<index>]` selects at most one object member value or at most one array element value.
 
 ~~~~ abnf
-index-selector      = "[" S (quoted-member-name / element-index) S "]"
+index-selector      = "[" S index-matcher S "]"
```

Comment:
The point of this approach is that we don't need `___-selector`s anymore.  There's no reason to define it since 1-length lists aren't used.

Reply:
I agree _syntactically_. Note that `index-selector` is deliberately unreferenced in the ABNF. It's meant to be a specific case of a child selector.

The reason for defining the index, slice, and filter selectors is to break the semantics of the child selector into manageable chunks which can then be combined in the semantics section of the child selector while avoiding defining the semantics of matchers except in the context of a selector.

Reply:
The problem is that you're still calling it an "index selector" when there is no such thing.  It's merely a child selector that happens to have a single index *picker.

Separately, we should keep using the placeholder until we have _all_ agreed on something else.

Reply:
> The problem is that you're still calling it an "index selector" when there is no such thing. It's merely a child selector that happens to have a single index *picker.

I have clarified that "index selector" is simply a variant of a "child selector". "index selector" is defined so it can be given semantics and then the semantics of the child selector can be constructed from the semantics of its syntactic variants. That's the gist of this approach: to gain the syntactic unification of PR #258 while preserving the current semantic structure.

> 
> Separately, we should keep using the placeholder until we have _all_ agreed on something else.

Fair enough. Actually I think, in this PR, there is no need for a special term so I've avoided it.

Reply:
I don't think we should create special-case selectors.  Instead, *pickers should have defined semantics that can interact with the defined semantics of the two selectors.
(Shortcuts are just a different way to write the selectors and don't have their own semantics.)